plugins/extractors/maxcompute/client/client.go

```diff
@@ -28,20 +30,33 @@ func New(conf config.Config) (*Client, error) {
 		return nil, err
 	}
 
+	properties, err := project.GetAllProperties()
+	if err != nil {
+		return nil, err
+	}
+	isSchemaEnabled := properties.Get("odps.namespace.schema") == "true"
+
 	return &Client{
-		client:  client,
-		project: project,
-		tunnel:  tunnelInstance,
+		client:          client,
+		project:         project,
+		tunnel:          tunnelInstance,
+		isSchemaEnabled: isSchemaEnabled,
 	}, nil
 }
 
-func (c *Client) ListSchema(context.Context) (schemas []*odps.Schema, err error) {
+func (c *Client) ListSchema(_ context.Context) (schemas []*odps.Schema, err error) {
 	err = c.project.Schemas().List(func(schema *odps.Schema, err2 error) {
 		if err2 != nil {
 			err = err2
 			return
 		}
-		schemas = append(schemas, schema)
+		if c.isSchemaEnabled {
+			if schema.Name() == "default" {
+				schemas = append(schemas, schema)
+			}
+		} else {
+			schemas = append(schemas, schema)
+		}
 	})
```

Comment:
if not enabled, let's not call the schema list at all, it might throw an error, so instead return []string{"default"}, nil directly before calling the list

Reply:
Should I change the function return type?, and handle it in Extract?

```
func (c *Client) ListSchema(_ context.Context) (schemas []string, err error) {
```

because won't able to return default without listing the schema, can only to return empty then

	```
if !c.isSchemaEnabled {
		return []*odps.Schema{}, nil
	}

	err = c.project.Schemas().List(func(schema *odps.Schema, err2 error) {
		if err2 != nil {
			err = err2
			return
		}
		schemas = append(schemas, schema)
	})

	return schemas, err
```